File: ass-ade-v1.1/tests/fixtures/rebuilt-out/bridges/swift/Sources/AssAdeBridge/a1_at_functions/ManifestLoader.swift
import Foundation

private func extractString(payload: String, prefix: String) -> String {
    guard let range = payload.range(of: prefix) else {
        return ""
    }
    let rest = payload[range.upperBound...]
    guard let firstQuote = rest.firstIndex(of: """) else {
        return ""
    }
    let afterFirst = rest[rest.index(after: firstQuote)...]
    guard let secondQuote = afterFirst.firstIndex(of: """) else {
        return ""
    }
    return String(afterFirst[..<secondQuote])
}

private func extractBool(payload: String, prefix: String) -> Bool {
    guard let range = payload.range(of: prefix) else {
        return false
    }
    return payload[range.upperBound...].trimmingCharacters(in: .whitespacesAndNewlines).hasPrefix("true")
}

private func extractStringArray(payload: String, prefix: String) -> [String] {
    guard let range = payload.range(of: prefix) else {
        return []
    }
    let rest = payload[range.upperBound...]
    guard let closing = rest.firstIndex(of: "]") else {
        return []
    }
    let segment = rest[..<closing]
    return segment.split(separator: """).enumerated().compactMap { index, item in
        index.isMultiple(of: 2) ? nil : String(item)
    }
}

func findRepoRoot(start: URL, manifestRel: String = defaultManifestRel) throws -> URL {
    var current = start
    while true {
        let candidate = current.appendingPathComponent(manifestRel)
        if FileManager.default.fileExists(atPath: candidate.path) {
            return current
        }
        let parent = current.deletingLastPathComponent()
        if parent.path == current.path {
            throw NSError(domain: "AssAdeBridge", code: 1, userInfo: [NSLocalizedDescriptionKey: "Could not find \(manifestRel)"])
        }
        current = parent
    }
}

func loadManifestPayload(start: URL, manifestRel: String = defaultManifestRel) throws -> String {
    let repoRoot = try findRepoRoot(start: start, manifestRel: manifestRel)
    return try String(contentsOf: repoRoot.appendingPathComponent(manifestRel), encoding: .utf8)
}

func loadBridgeManifest(start: URL) throws -> BridgeManifestSummary {
    let payload = try loadManifestPayload(start: start)
    return BridgeManifestSummary(
        schema: extractString(payload: payload, prefix: ""schema": "),
        bridgeReady: extractBool(payload: payload, prefix: ""bridge_ready": "),
        bridgeLanguages: extractStringArray(payload: payload, prefix: ""bridge_languages": ["),
        pythonBridgeCommand: extractStringArray(payload: payload, prefix: ""python_bridge_command": [")
    )
}

File: tests/v11/fixtures/rebuilt-out/bridges/rust/src/a1_at_functions/manifest_loader.rs
use std::fs;
use std::path::PathBuf;

use crate::a0_qk_constants::bridge_contract::{BridgeManifestSummary, DEFAULT_MANIFEST_REL};

fn extract_string(payload: &str, prefix: &str) -> String {
    payload
        .split(prefix)
        .nth(1)
        .and_then(|rest| rest.split('"').nth(1))
        .unwrap_or_default()
        .to_string()
}

fn extract_bool(payload: &str, prefix: &str) -> bool {
    payload
        .split(prefix)
        .nth(1)
        .map(|rest| rest.trim_start().starts_with("true"))
        .unwrap_or(false)
}

fn extract_string_array(payload: &str, prefix: &str) -> Vec<String> {
    let Some(rest) = payload.split(prefix).nth(1) else {
        return Vec::new();
    };
    let Some(segment) = rest.split(']').next() else {
        return Vec::new();
    };
    segment
        .split('"')
        .skip(1)
        .step_by(2)
        .map(str::to_string)
        .collect()
}

pub fn find_repo_root(start: PathBuf) -> Result<PathBuf, String> {
    let mut current = start;
    loop {
        let candidate = current.join(DEFAULT_MANIFEST_REL);
        if candidate.is_file() {
            return Ok(current);
        }
        let Some(parent) = current.parent() else {
            return Err(format!("could not find {}", DEFAULT_MANIFEST_REL));
        };
        current = parent.to_path_buf();
    }
}

pub fn load_manifest_payload(start: PathBuf) -> Result<String, String> {
    let repo_root = find_repo_root(start)?;
    let manifest_path = repo_root.join(".ass-ade/bridges/bridge_manifest.json");
    fs::read_to_string(manifest_path).map_err(|err| err.to_string())
}

pub fn load_manifest(start: PathBuf) -> Result<BridgeManifestSummary, String> {
    let payload = load_manifest_payload(start)?;
    Ok(BridgeManifestSummary {
        schema: extract_string(&payload, "\"schema\": "),
        bridge_ready: extract_bool(&payload, "\"bridge_ready\": "),
        bridge_languages: extract_string_array(&payload, "\"bridge_languages\": ["),
        python_bridge_command: extract_string_array(&payload, "\"python_bridge_command\": ["),
    })
}
